runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -54,49 +58,119 @@ void waitForWorkflow(UUID workflow, ManagementClient client, Duration timeout,
+                        // Validation has succeeded
                         return;
                     }
+                    log.warn("runWorkflow: failed verification for {} on try {}", req.getId(), y);
                 }
-                log.warn("addNode: Couldn't find {} in {}, retrying...", endpointToAdd, newLayout, x);
-
-            } catch (NetworkException | TimeoutException  e) {
-                log.warn("addNode: Error while requesting to add {}", endpointToAdd, e);
+            } catch (NetworkException | TimeoutException e) {
+                log.warn("runWorkflow: Error ", e);
                 continue;
             }
+            log.warn("runWorkflow: Failed to run {} on try {}", req == null ? null : req, x);
         }
 
-        log.warn("addNode: Failed to add node {}", endpointToAdd);
-        throw new WorkflowException("Failed to add " + endpointToAdd);
+        throw new WorkflowResultUnknownException();
+    }
+
+    /**
+     * Remove a node from the cluster.
+     *
+     * @param endpointToRemove Endpoint of the node to be removed from the cluster.
+     * @param retry            the number of times to retry a workflow if it fails
+     * @param timeout          total time to wait before the workflow times out
+     * @param pollPeriod       the poll interval to check whether a workflow completed or not
+     * @throws WorkflowResultUnknownException when the side affect of the operation
+     *                                        can't be determined
+     */
+    public void removeNode(@Nonnull String endpointToRemove, int retry,
+                           @Nonnull Duration timeout, @Nonnull Duration pollPeriod) {
+        runWorkflow(layout -> {
+
+                    // Select an orchestrator that is not on a node that is not being removed
+                    // and send a remove node request
+                    layout = runtime.getLayoutView().getLayout();
+                    List<String> layoutServers = new ArrayList(layout.getLayoutServers());
+                    layoutServers.remove(endpointToRemove);
+                    if (layoutServers.isEmpty()) {
+                        throw new WorkflowException("Can't remove node from a single node cluster");
+                    }
+                    String orchestratorEndpoint = layoutServers.get(layoutServers.size() - 1);
+                    ManagementClient client = runtime.getRouter(orchestratorEndpoint)
+                            .getClient(ManagementClient.class);
+                    CreateWorkflowResponse resp = client.removeNode(endpointToRemove);
+                    log.info("removeNode: Requested to remove {} on orchestrator {}, layout {}",
+                            endpointToRemove, orchestratorEndpoint, layout);
+                    return new Request(resp.getWorkflowId(), client);
+                },
+
+                layout -> {
+                    // Verify that the node doesn't exist in the new layout
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)